ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -0,0 +1,387 @@
+    log.info(
+        f"New images for upgrade: "
+        f"{sorted(new_images_to_upgrade)}"
+    )
+    unchanged_images = old_csv_images.intersection(new_csv_images)
+    log.info(
+        f"Unchanged images after upgrade: "
+        f"{sorted(unchanged_images)}"
+    )
+    return (
+        old_images_for_upgrade,
+        new_images_to_upgrade,
+        unchanged_images,
+    )
+
+
+def verify_image_versions(old_images, upgrade_version):
+    """
+    Verify if all the images of OCS objects got upgraded
+
+    Args:
+        old_images (set): set with old images
+        upgrade_version (packaging.version.Version): version of OCS
+
+    """
+    number_of_worker_nodes = len(get_typed_nodes())
+    osd_count = get_osd_count()
+    verify_pods_upgraded(old_images, selector=constants.OCS_OPERATOR_LABEL)
+    verify_pods_upgraded(old_images, selector=constants.OPERATOR_LABEL)
+    # in 4.3 app selector nooba have those pods: noobaa-core-ID, noobaa-db-ID,
+    # noobaa-operator-ID but in 4.2 only 2: noobaa-core-ID, noobaa-operator-ID
+    nooba_pods = 2 if upgrade_version < parse_version('4.3') else 3
+    verify_pods_upgraded(
+        old_images, selector=constants.NOOBAA_APP_LABEL, count=nooba_pods
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_CEPHFSPLUGIN_LABEL,
+        count=number_of_worker_nodes,
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_CEPHFSPLUGIN_PROVISIONER_LABEL,
+        count=2
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_RBDPLUGIN_LABEL,
+        count=number_of_worker_nodes,
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_RBDPLUGIN_PROVISIONER_LABEL,
+        count=2
+    )
+    verify_pods_upgraded(old_images, selector=constants.MON_APP_LABEL, count=3)
+    verify_pods_upgraded(old_images, selector=constants.MGR_APP_LABEL)
+    # OSD upgrade have timeout 10mins for new attempt if cluster is not health.
+    # https://bugzilla.redhat.com/show_bug.cgi?id=1840729 setting timeout for
+    # 12.5 minutes per OSD
+    verify_pods_upgraded(
+        old_images, selector=constants.OSD_APP_LABEL, count=osd_count,
+        timeout=750 * osd_count,
+    )
+    verify_pods_upgraded(old_images, selector=constants.MDS_APP_LABEL, count=2)
+    if config.ENV_DATA.get('platform') == constants.VSPHERE_PLATFORM:
+        verify_pods_upgraded(
+            old_images, selector=constants.RGW_APP_LABEL, count=1
+        )
+
+
+class OCSUpgrade(object):
+    """
+    OCS Upgrade helper class
+
+    """
+    def __init__(self, namespace, version_before_upgrade, ocs_registry_image, upgrade_in_current_source):
+        self.namespace = namespace
+        self.version_before_upgrade = version_before_upgrade
+        self.ocs_registry_image = ocs_registry_image
+        self.upgrade_in_current_source = upgrade_in_current_source
+
+    @property
+    def get_version_before_upgrade(self):
```

Comment:
As this is a property and not a getter, this should not be named `get_version_before_upgrade` but `version_before_upgrade`. It is not called like a function but as an attribute so I don't think `get_` prefix should be used. It could look like this:
```
    @property
    def version_before_upgrade(self):
        return self._version_before_upgrade
```
The same for property `get_ocs_registry_image`.